### PR TITLE
httpsrv client, add threading support

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: twatbot.py [hunk 1]
```diff
@@ -392,33 +392,53 @@ def serve_loop(ip, port, done):
 	from httpsrv import HttpSrv
 	hs = HttpSrv(ip, port)
 	hs.setup()
+	client_threads = []
 	while not done.is_set():
+		ctrm = []
```
Comment (rofl0r) [inline twatbot.py]: can you move this below the wait_client() too ? it belongs there, logically

File: twatbot.py [hunk 2]
```diff
@@ -392,33 +392,53 @@ def serve_loop(ip, port, done):
...
-
-			r, redir = render_site(vars)
-			if redir is not "":
-				c.redirect(redir)
-			else:
-				c.send(200, "OK", r)
-		elif not '..' in req['url'] and file_exists(os.getcwd() + req['url']):
-			c.serve_file(os.getcwd() + req['url'])
+
+		for ct, ct_done in client_threads:
```
Comment (rofl0r) [inline twatbot.py]: this stuff should be done *after* the client thread is started :)

File: twatbot.py [hunk 3]
```diff
@@ -392,33 +392,53 @@ def serve_loop(ip, port, done):
...
+				ct.join()
+
+		if len(ctrm):
+			client_threads = [ x for x in client_threads if not x in ctrm ]
+
+		evt_done = threading.Event()
+		cthread = threading.Thread(target=httpsrv_client_thread, args=(c,evt_done))
+		cthread.daemon = True
+		client_threads.append((cthread, evt_done))
+		cthread.start()
```
Comment (rofl0r) [inline twatbot.py]: if you move this start() one up, you could check the existing list of threads for terminated threads to reap, and then append our new thread

**Comment by ghost** [comment]
PR moved to new_branch :-)

**Pull request closed without merging**